chapters/mapping_data_to_shaders.md

```diff
@@ -0,0 +1,308 @@
+# Mapping Data to Shaders
+
+> Note: All SPIR-V assembly was generated with glslangValidator
+
+This chapter goes over how to [interface Vulkan with SPIR-V](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/html/vkspec.html#interfaces) in order to map data. Using the `VkDeviceMemory` objects allocated from `vkAllocateMemory`, it is up to the application to properly map the data from Vulkan in a way that the SPIR-V shader understands how to consume it correctly.
+
+In core Vulkan, there are 4 fundamental ways to map data from your Vulkan application to interface with SPIR-V:
+  - [Input Attributes](#input-attributes)
+  - [Descriptors](#descriptors)
+  - [Push Constants](#push-constants)
+  - [Specialization Constants](#specialization-constants)
```

Comment:
I think you need to mention buffer pointers. They’re a powerful tool for a range of use cases, and not mentioning them feels wrong to me.